### PR TITLE
Fix dependency between toolbar and clipboard

### LINKED ISSUE #654: Remove dependency on toolbar from clipboard
## Are you reporting a feature or a bug?

Task

## Explanation

Currently clipboard plugin depends on two other plugins: notification and toolbar. However it seems that toolbar-specific functions are used only to disallow dropping files on it. Therefore dropping it as a dependency seems like a relatively simple task.

It's kinda-sorta followup of #491.

### PR DESCRIPTION
**Pull request by msamsel**
## What is the purpose of this pull request?

Bug fix.

## Does your PR contain necessary tests?

yes

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

* Remove dependancy on toolbar in clipboard plugin
* Improvement in Link and Table plugin to register proper ACF rules even when toolbar is not present. This is required to pass tests (tableselection, autoemebed).

Close #654

### CODE REVIEW

File: plugins/link/plugin.js [hunk 1]
```diff
@@ -49,20 +49,25 @@
```
Comment (Comandeer) [inline plugins/link/plugin.js]: It doesn't seem to be part of the fix for #654. If #654 does not make sense without #678, it should be blocked on it.

What's more, if #678 is resolved, these changes will become obsolete.

**Comment by msamsel** [comment]
I haven't thought in such way. I removed commits which change `link` and `table` plugin, but I had to modify `autoembed` and `tableselection` tests instead.

File: plugins/clipboard/plugin.js [hunk 2]
```diff
@@ -1306,8 +1306,13 @@
 			// -------------- DRAGOVER TOP & BOTTOM --------------
 
 			// Not allowing dragging on toolbar and bottom (http://dev.ckeditor.com/ticket/12613).
-			clipboard.preventDefaultDropOnElement( top );
-			clipboard.preventDefaultDropOnElement( bottom );
+			// Remove dependacy on toolbar (#654).
```
Comment (Comandeer) [inline plugins/clipboard/plugin.js]: Please fix typo here and in other places (it's "dependency", not "dependancy").

**Comment by mlewand** [comment]
@Comandeer actually you're right. Removing this dep would break many dependent plugin, unaware of this long standing issue. Given that I'm all for blocking this issue until #678 is done.

**Pull request closed without merging**